libs/ledger-live-common/src/families/aptos/getFeesForTransaction.test.ts

```diff
@@ -0,0 +1,221 @@
+            expirationTimestampSecs: 5,
+          },
+          errors: {
+            expirationTimestampSecs: ["TRANSACTION_EXPIRED"],
+          },
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe("with vm_status as INSUFFICIENT_BALANCE", () => {
+      it("should return a fee estimation object", async () => {
+        simulateTransaction = jest.fn(() => [
+          {
+            success: false,
+            vm_status: ["INSUFFICIENT_BALANCE"],
+            expiration_timestamp_secs: 5,
+          },
+        ]);
+
+        const account = createFixtureAccount();
+        const transaction = createTransaction();
+        const aptosClient = new AptosAPI(account.currency.id);
+
+        transaction.amount = new BigNumber(1);
+        account.xpub = "xpub";
+        account.spendableBalance = new BigNumber(100000000);
+
+        const result = await getFee(account, transaction, aptosClient);
+
+        const expected = {
+          fees: new BigNumber(20301),
+          estimate: {
+            maxGasAmount: "201",
+            gasUnitPrice: "101",
+            sequenceNumber: "123",
+            expirationTimestampSecs: 5,
+          },
+          errors: {},
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe("with vm_status as DUMMY_STATE", () => {
+      it("should return a fee estimation object", () => {
+        simulateTransaction = jest.fn(() => [
+          {
+            success: false,
+            vm_status: ["DUMMY_STATE"],
+            expiration_timestamp_secs: 5,
+          },
+        ]);
+
+        const account = createFixtureAccount();
+        const transaction = createTransaction();
+        const aptosClient = new AptosAPI(account.currency.id);
+
+        transaction.amount = new BigNumber(1);
+        account.xpub = "xpub";
+        account.spendableBalance = new BigNumber(100000000);
+
+        expect(async () => {
+          await getFee(account, transaction, aptosClient);
+        }).rejects.toThrow("Simulation failed with following error: DUMMY_STATE");
+      });
+    });
+  });
+
+  describe("when using getEstimatedGas", () => {
+    describe("when key not in cache", () => {
+      it("should return cached fee", async () => {
+        const account = createFixtureAccount();
+        const transaction = createTransaction();
+        const aptosClient = new AptosAPI(account.currency.id);
+
+        const result = await getEstimatedGas(account, transaction, aptosClient);
+
+        const expected = {
+          errors: {},
+          estimate: {
+            expirationTimestampSecs: "",
+            gasUnitPrice: "101",
+            maxGasAmount: "201",
+            sequenceNumber: "123",
+          },
+          fees: new BigNumber("20301"),
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe("when key is in cache", () => {
+      it("should return cached fee", async () => {
```

Comment:
We have two "should return cached fee" clauses

here we need to change mock value on the second time, so we make sure we return the cached value and not current value

eg. we call the first time, then we change the mock value, we call it the second time, it should return the first mock value and not the second